templates/pyproject.toml

```diff
@@ -0,0 +1,205 @@
+[build-system]
+# Defined by PEP 518
+requires = [
+# Some packages will require extra additions to this list. E.g "Cython" or "Numpy"
+  "setuptools>=61",
+  "setuptools_scm[toml]>=7",
+  "wheel"
+]
+# Defined by PEP 517
+build-backend = "setuptools.build_meta"
+
+[project]
+authors = [
+    {name = "<PACKAGE_NAME> Contributors", email = "scitools.pub@gmail.com"},
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: THIS VARIES BY REPOSITORY",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: SUPPORTED PYTHON VERSIONS",
+    # e.g "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Atmospheric Science",
+    "Topic :: Scientific/Engineering :: VARIES BY REPOSITORY",
+]
+dynamic = [
+    "dependencies",
+    "readme",
+    "version",
+]
+description = "A powerful, format-agnostic, community-driven Python package for analysing and visualising Earth science data"
+keywords = [
+    "<PACKAGE_KEYWORDS>",
+]
+license = {text = "BSD-3-Clause"}
+name = "scitools-<PACKAGE_NAME>" #e.g "scitools-iris"
+requires-python = "<MIN_PYTHON_VERSION>"
+
+[project.urls]
+Code = "https://github.com/SciTools/<PACKAGE_NAME>"
+Discussions = "https://github.com/SciTools/<PACKAGE_NAME>/discussions" # If discussions are enabled.
+Documentation = "https://scitools-iris.readthedocs.io/" # If you have a documentation site.
+Issues = "https://github.com/SciTools/<PACKAGE_NAME>/issues" # If issues are enabled.
+
+[tool.check-manifest]
+# Ignores here will vary by repository.
+ignore = []
+
+[tool.codespell]
+ignore-words-list = []
+
+[tool.coverage.report]
+exclude_lines = [
+    "pragma: no cover",
+    "def __repr__",
+    "if __name__ == .__main__.:",
+]
+
+[tool.coverage.run]
+branch = true
+include = [
+    "<PACKAGE_NAME>/*",
+]
+omit = [
+    "<PATH_TO_TESTS>/*", # e.g "cf_units/tests/*" or "lib/iris/tests/*"
+]
+
+[tool.mypy]
+disable_error_code = [] # Include any error codes that you want to ignore
+enable_error_code = [
+    "ignore-without-code",
+    # "redundant-expr",		# TODO: Add back in when above ignores fixed
+    "truthy-bool",
+]
+exclude = [
+]
+ignore_missing_imports = true
+strict = true
+warn_unreachable = true
+warn_unused_configs = true
+warn_unused_ignores = true
+
+[tool.numpydoc_validation]
+# We have agreed a that some of the numpydoc rules that we are going to ignore.
+# Our repositories are going to have extra rules that they are ignoring that is expected, but they are not included here.
+checks = [
```

Comment:
The URL somewhere for where people can look these up?

https://numpydoc.readthedocs.io/en/latest/validation.html#validation-checks